core/internal/cltest/cltest.go

```diff
@@ -1683,3 +1687,48 @@ func ClearDBTables(t *testing.T, db *sqlx.DB, tables ...string) {
 	err = tx.Commit()
 	require.NoError(t, err)
 }
+
+// CopyFlagSetFromAction takes the given flagSet and applies the actions flag to it.
+// `parentCommand` will filter the app commands and only applies the flags if the command/subcommand has a parent with that name, if left empty no filtering is done
+func CopyFlagSetFromAction(action interface{}, flagSet *flag.FlagSet, parentCommand string) {
+	cliApp := cmd.Client{}
+	app := cmd.NewApp(&cliApp)
+
+	foundName := parentCommand == ""
+
+	for _, command := range app.Commands {
+		flags := recursiveFindFlagsWithName(action, command, parentCommand, foundName)
+
+		if flags != nil {
+			for _, flag := range flags {
+				flag.Apply(flagSet)
+			}
+		}
+	}
+
+}
+
+func recursiveFindFlagsWithName(action interface{}, command cli.Command, parent string, foundName bool) []cli.Flag {
+
+	if command.Action != nil {
+		if getFunctionName(action) == getFunctionName(command.Action) && foundName {
```

Comment:
It looks like `getFunctionName(action)` gets called a lot. Maybe that could be computed just once from `CopyFlagSetFromAction` and then passed in here as `actionName string` or something?

Reply:
Fixed in [0b548f9](https://github.com/smartcontractkit/chainlink/pull/8173/commits/0b548f91b35e5fcb6c95002d26530e1faf342b43)